docs/BUILDING.md

```diff
@@ -290,3 +294,18 @@ To run the tests, execute the test binary in the build directory:
 [Homebrew]: https://brew.sh/
 [Visual Studio]: https://visualstudio.microsoft.com/vs/
 [Nix]: https://nixos.org/manual/nix/stable/
+
+---
+
+### Troubleshooting
+
+#### Language server cannot find some files
+
+If your C++ language server cannot find certain headers, you may need to
+configure CMake to export the compile commands by using
+`-DCMAKE_EXPORT_COMPILE_COMMANDS=1`. E.g.:
```

Comment:
I think this is an odd place to put this documentation. The solution is already documented on clangd's website: https://clangd.llvm.org/installation#compile_commandsjson

Do you still think it's worth including this language-server-specific documentation in quick-lint-js' docs?

Reply:
I agree that it is not ideal, but I thought it would make sense mostly because it alters the build command, so some people can find odd that "I executed the command to build quick-lint and it didn't work out-of-the-box"

I think it is also worth noticing that some users don't know what language-server they are using behind the curtain. As someone who is new to C++, I didn't know my editor was automatically using clangd, so it took me quite some research to find out how to get these headers to be found properly

Do you think it would make more sense if, instead of pointing exact flags, I changed the docs to make it more clear that there are some extra configurations the developer can do? Something like:

> Run the following command to create a directory called `build`:
>
>    $ mkdir build ; cd build ; cmake -G Ninja -DCMAKE_BUILD_TYPE=Debug .. ; cd "$OLDPWD"
>
> Notice that, depending on your development environment, you may need some extra configurations. Here are some places to get you started:
> - For clangd users: \<insert docs you mentioned>

Reply:
> I thought it would make sense mostly because it alters the build command

That's a good point.

> some people can find odd that "I executed the command to build quick-lint and it didn't work out-of-the-box"

In that case, why not include `=DCMAKE_EXPORT_COMPILE_COMMANDS=1` in all documented `cmake` invocations, removing the need for this troubleshooting section? Is there any downside to enabling `=DCMAKE_EXPORT_COMPILE_COMMANDS=1` in all builds?

Reply:
I didn't find any downsides that seem relevant to quick-lint in [cmake docs](https://cmake.org/cmake/help/latest/variable/CMAKE_EXPORT_COMPILE_COMMANDS.html), so I believe including the flag in all invocations will be a better solution indeed